ehrql/backends/tpp.py

```diff
@@ -22,6 +25,62 @@ class TPPBackend(BaseBackend):
         ehrql.tables.beta.smoketest,
     ]
 
+    include_t1oo = False
+
+    def modify_dsn(self, dsn):
+        """
+        Removes the `opensafely_include_t1oo` parameter if present and uses it to set
+        the `include_t1oo` attribute accordingly
+        """
+        parts = parse.urlparse(dsn)
+        params = parse.parse_qsl(parts.query, keep_blank_values=True)
+
+        include_t1oo_values = []
+        other_params = []
+        for key, value in params:
+            if key == "opensafely_include_t1oo":
+                include_t1oo_values.append(value)
+            else:
+                other_params.append((key, value))
+
+        if len(include_t1oo_values) == 1:
+            self.include_t1oo = include_t1oo_values[0].lower() == "true"
+        elif len(include_t1oo_values) != 0:
+            raise ValueError(
+                "`opensafely_include_t1oo` parameter must not be supplied more than once"
```

Comment:
Do we care if it's supplied more than once if the value is the same each time?

Reply:
Interestingly, I originally used a `set` here for exactly this behaviour. But then I thought we're in control of this URL and there's no good reason we should be setting it twice and if we do the resulting URL is likely to be confusing, so it would be better to be strict.